### PR TITLE
clippy: ignore false map_clone warning

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

some clippy changes for Rust v1.78.0. (#1309)

it complains about this line but which is an arc clone.
https://github.com/anza-xyz/agave/blob/718101183271a0b75f307d86bde48721051ca5e9/bucket_map/src/bucket_storage.rs#L521

#### Summary of Changes

ignore false map_clone warning.

### CODE REVIEW

File: bucket_map/src/bucket_storage.rs [hunk 1]
```diff
@@ -501,6 +501,7 @@ impl<O: BucketOccupied> BucketStorage<O> {
     }
 
     /// allocate a new bucket, copying data from 'bucket'
+    #[allow(clippy::map_clone)] // https://github.com/rust-lang/rust-clippy/issues/12560
```
Comment (brooksprumo) [inline bucket_map/src/bucket_storage.rs]: Thanks for finding the GH issue for this clippy bug! One question, can this `#[allow]` go down on the actual call to `Arc::clone()`, or does it need to be on the function itself?

Comment (yihau) [reply bucket_map/src/bucket_storage.rs]: yeah. I think we can do
```diff
     /// allocate a new bucket, copying data from 'bucket'
-    #[allow(clippy::map_clone)] // https://github.com/rust-lang/rust-clippy/issues/12560
     pub fn new_resized(
         drives: &Arc<Vec<PathBuf>>,
         max_search: MaxSearch,
@@ -518,6 +517,7 @@ impl<O: BucketOccupied> BucketStorage<O> {
             capacity,
             max_search,
             Arc::clone(stats),
+            #[allow(clippy::map_clone)] // https://github.com/rust-lang/rust-clippy/issues/12560
             bucket
                 .map(|bucket| Arc::clone(&bucket.count))
                 .unwrap_or_default(),
```

do we prefer this one? let me know!

Comment (brooksprumo) [reply bucket_map/src/bucket_storage.rs]: IMO yes, assuming clippy is still happy!

Comment (yihau) [reply bucket_map/src/bucket_storage.rs]: yeah. clippy is a nice guy :trollface: fixed: [facf565](https://github.com/anza-xyz/agave/pull/1243/commits/facf5654dca2cc8890ea6afd976af8105a1439f2)

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Pull request merged**